_pages/dev/tag-management-system-examples.md

```diff
@@ -0,0 +1,112 @@
+# Tag Management System Examples
+
+This doc will briefly explain how to setup some of the most useful TMS solution on the market. For more details on the configuration and extensibility part, please check the main [TMS docs](tag-management-system.md).
+
+## Google Tag Manager (GTM)
+
+Spartacus offers an out of the box support for GTM, and in order to configure it you need to first include a code snippet provided by GTM to your `index.html`. The code snippet will look something like this:
+
+```html
+<!-- Google Tag Manager -->
+<script>
+  (function (w, d, s, l, i) {
+    w[l] = w[l] || [];
+    w[l].push({ "gtm.start": new Date().getTime(), event: "gtm.js" });
+    var f = d.getElementsByTagName(s)[0],
+      j = d.createElement(s),
+      dl = l != "dataLayer" ? "&l=" + l : "";
+    j.async = true;
+    j.src = "https://www.googletagmanager.com/gtm.js?id=" + i + dl;
+    f.parentNode.insertBefore(j, f);
+  })(window, document, "script", "dataLayer", "GTM-XXXX");
+</script>
+<!-- End Google Tag Manager -->
+```
+
+For more details on this, please check the [official docs](https://developers.google.com/tag-manager/quickstart).
+
+## Google Analytics (GA)
+
+As both GA and GTM push events in the same fashion (they both push the events to the `dataLayer`), you can still leverage our out-of-the-box support for GTM, but include a different code snippet to your `index.html`.
+
+As per the [official GA docs](https://developers.google.com/analytics/devguides/collection/gtagjs), the code snippet should looks something like:
+
+```html
+<!-- Global site tag (gtag.js) - Google Analytics -->
+<script
+  async
+  src="https://www.googletagmanager.com/gtag/js?id=GA_MEASUREMENT_ID"
+></script>
+<script>
+  window.dataLayer = window.dataLayer || [];
+  function gtag() {
+    dataLayer.push(arguments);
+  }
+  gtag("js", new Date());
+
+  gtag("config", "GA_MEASUREMENT_ID");
+</script>
+```
+
+## GTM and GA
+
+Running both GTM and GA _at the same time_ is possible if you have a need for such a use case, although you should be aware of all the potential clashes of doing so.
```

Comment:
I'm afraid you made me curious on the why :) Can you elaborate a little? In slack is fine as well. 

I was actually expecting a very different approach here, so this one is interesting to learn! 

I was assuming the following:
1. You generate a GA tag with GTM, which adds the script tag (including specific ga IDs). 
2. The GA integration would work in a traditional way, meaning that the GA events must be build separately. here's where my knowledge felt short; using the dataLayer for the same is a big win (and makes so much sense). 

Perhaps adding GA through GTM is still a valid option and would simplify the setup.

Reply:
Started discussion on Slack